### ISSUE TITLE
File Manager not working with PHP < 5.4

### ISSUE THREAD

**Issue by travisbotello** [wolfcms/wolfcms#631]
Hello,

I just updated two of my Wolf CMS installations, and on both the administration subpage is not working anymore. I was also able to reproduce this bug on a clean 0.8.3.1 installation.

What steps will reproduce the problem?
- Just log in to the manager 
- Go to Administration 
- You will see a subpage with two tabs 'Plugins/Settings' 

What is the expected output? What do you see instead?
- the 'You are currently logged in as Administrator | Log Out | View Site' subnavigation disappears
- the active status on both tabs is not working
- you can't switch between plugins and settings. So there is no chance to get to the system setttings anymore
- the plugin list is short. A lot of plugins are missing.

What version of Wolf CMS are you using?
- 0.8.3.1

What PHP version are you using?
- 5.3.26

On what operating system are you using Wolf CMS?
- CentOS 5.11

What database are you using? (please include the version number as well)
- MySQL 5.5.30

**Comment by project79** [comment]
Strange. I see you are not the only one. There was one more guy on forum that has the same problem.

I just upgraded (locally) and I don't have this issue. PHP 5.6.11 on Xampp.

**Comment by project79** [comment]
Please check index.php for version number. It should be: define('CMS_VERSION', '0.8.3');

**Comment by travisbotello** [comment]
Correct, it's: define('CMS_VERSION', '0.8.3');

Please note that I am also having this issue with a clean 0.8.3 installation.

**Comment by project79** [comment]
Then it must me something with the system (I'm guessing here). Let me update my site on linux and I'll get back.

**Comment by djgprv** [comment]
I use usbweb php5.3 and centOS php5.3.3 file meneger plugin does not work
too. Version in index gile is 083 but steill ask for an update.

**Comment by project79** [comment]
Ok, Did it and no problem with Administration.
It shared linux server, PHP 5.3.28, MySQL 5.5.42 with Apache 2.4.

Try to overwrite /wolf and index.php again. Maybe something got screwed the first time.

**Comment by travisbotello** [comment]
I don't think that it has to do with the system, PHP Version, etc.

Obviously something was changed here. In Wolf CMS 0.8.2 there was no url update when switching the tabs on the Administration subpage. In Wolf CMS 0.8.3.1 the URL changes to #settings and #plugins ... maybe there is something wrong

**Comment by travisbotello** [comment]
@project79 I doesn't make any sense to overwrite again since I am using a fresh and clean install of 0.8.3.1. So there was no update process at all...

**Comment by project79** [comment]
I don't have #hash changes with #plugin and #setting. It's always #setting!

Please see Javascript console on your browser, just to rule that out.

**Comment by travisbotello** [comment]
/admin/setting#plugins vs. /admin/setting#settings

These are the hash changes that occur. No javascript errors in the console!

**Comment by project79** [comment]
@djgprv - Ah that. That is because there is no version 0.8.3.1, only 0.8.3. Temporarily turn off updates until that does not sorts out. We had similar problems before too.

**Comment by project79** [comment]
Or add .1 in a versin number in index.php and it will work fine afterwards.

**Comment by travisbotello** [comment]
It's very interesting that the <div id="settings"> is completely missing in the 0.8.3.1 installation. So there's no way to switch the tabs...

**Comment by project79** [comment]
The fix for 0.8.3  was only in Filemanager and Login Controller so everything else should stay as it was.

Everything is there: https://github.com/wolfcms/wolfcms/blob/master/wolf/app/views/setting/index.php

Try to overwrite this one with the one you have.

**Comment by travisbotello** [comment]
Okay...well something seems to be messed up here. I will compare the files and let you know why the settings tab is missing.

**Comment by djgprv** [comment]
Strange, downloaded the current version again and upgrade a clean install updates 082. Works fine. File manager as well. I'll check again the current websites operating on 082.

**Comment by travisbotello** [comment]
Problem identified:

In https://github.com/wolfcms/wolfcms/blob/master/wolf/app/views/setting/index.php line 56-61:

```
<h4>`
<?php
 if (isset($loaded_plugins[$plugin->id]) && Plugin::hasDocumentationPage($plugin->id) )
    '<a href="'.get_url('plugin/'.$plugin->id.'/documentation').'">'.$plugin->title.'</a>';
else
    echo $plugin->title;
?>
```

This breaks the code. So my Administration page is cut off after `<h4>`. This explains why the subnav, and the tab switching is not working.

@project79 You were right. Nothing was changed here from 0.8.2 to 0.8.3.1 however this code doesn't lead to any problems in my 0.8.2 installation while it breaks in 0.8.3.1

Any ideas?

**Comment by travisbotello** [comment]
Update: The file manager is causing the problem. When I disable the file manager everything is working as expected. When I enable the file manager the backend breaks again...

**Comment by travisbotello** [comment]
Problem solved. I created a pull request:

https://github.com/wolfcms/wolfcms/commit/fb24d2984fe94361b8f8e3ca50c418e325952b47

The second argument in `if (in_array($ext, ['php', 'php3', 'php4', 'inc'])) {` was causing the following php error msg: *Parse error: syntax error, unexpected '[...*

I changed the code to store the file extensions in the variable $ext_match before the if clause and now everything is working as expected! See pull request for details.

**Comment by mvdkleijn** [comment]
Ah! Apologies.... [...] is a PHP 5.4 notation for arrays! So only users with PHP < 5.4 will have this issue.

**Comment by travisbotello** [comment]
That's correct!

**Comment by mvdkleijn** [comment]
Fixed in master and 0.8.3.1 download.
